server/metadata/database.go

```diff
@@ -23,3 +25,59 @@ type DatabaseMetadata struct {
 func (dm *DatabaseMetadata) SetDatabaseId(id uint32) {
 	dm.Id = id
 }
+
+const (
+	MainBranch          = "main"
+	BranchNamePrefix    = "$branch$_"
+	BranchNameSeparator = "_"
+)
+
+type DatabaseName struct {
+	db     string
+	branch string
+}
+
+func NewDatabaseNameWithBranch(db string, branch string) *DatabaseName {
+	if branch == MainBranch {
+		branch = ""
+	}
+	return &DatabaseName{
+		db:     db,
+		branch: branch,
+	}
+}
+
+func NewDatabaseName(key string) *DatabaseName {
+	if !strings.HasPrefix(key, BranchNamePrefix) {
+		return NewDatabaseNameWithBranch(key, "")
+	}
+	s := strings.SplitN(key, BranchNameSeparator, 3)
+	return NewDatabaseNameWithBranch(s[1], s[2])
+}
+
+func (b *DatabaseName) Key() string {
```

Comment:
just call the method `Name`? Key would be confusing in the context of database name.